### PR TITLE
tests: roachtest that creates overload when doing intent resolution

### PR DESCRIPTION
**Pull request by sumeerbhola**
Informs #97108

Epic: CRDB-25458

Release note: None

### CODE REVIEW

**Comment by sumeerbhola** [comment]
L0 sublevels goes over the threshold due to the lack of AC
Screenshot 2023-08-17 at 12 40 21 PM

File: pkg/cmd/roachtest/tests/admission_control_intent_resolution.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+// intents. When intent resolution is not subject to admission control, the
+// LSM gets overloaded and has > 50 sub-levels.
+func registerIntentResolutionOverload(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:      "admission-control/intent-resolution",
+		Owner:     registry.OwnerAdmissionControl,
+		Timeout:   time.Hour,
+		Benchmark: true,
+		Tags:      registry.Tags(`weekly`),
+		// Second node is solely for Prometheus.
```
Comment (irfansharif) [inline pkg/cmd/roachtest/tests/admission_control_intent_resolution.go:40]: We don't need to BTW - even running prometheus on the same node, or not at all, is ok. These runs happen on GCE and we now have that global prometheus instance that knows how to scrape every endpoint.

File: pkg/cmd/roachtest/tests/admission_control_intent_resolution.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+// that creates a huge number of intents, and then commits to resolve those
+// intents. When intent resolution is not subject to admission control, the
+// LSM gets overloaded and has > 50 sub-levels.
+func registerIntentResolutionOverload(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:      "admission-control/intent-resolution",
+		Owner:     registry.OwnerAdmissionControl,
+		Timeout:   time.Hour,
+		Benchmark: true,
+		Tags:      registry.Tags(`weekly`),
```
Comment (irfansharif) [inline pkg/cmd/roachtest/tests/admission_control_intent_resolution.go:39]: Run nightly for a while to shake out flakes before reducing to weekly frequency?

**Review by irfansharif — APPROVED** [review]

**Review by sumeerbhola — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/108916#-:-Nc5LwtB0nw51b7ViY5o:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @herkolategan, and @irfansharif)

___
*[`pkg/cmd/roachtest/tests/admission_control_intent_resolution.go` line 39 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/108916#gh-1297699625:-Nc5LMCACQBK79eWzL5C:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/ee2f24d5d9a7cda5b08b271de382efa9ad845155/pkg/cmd/roachtest/tests/admission_control_intent_resolution.go#L39)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

Run nightly for a while to shake out flakes before reducing to weekly frequency?
</blockquote></details>

Done
___
*[`pkg/cmd/roachtest/tests/admission_control_intent_resolution.go` line 40 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/108916#gh-1297698769:-Nc5LP-t8jCN-LLCEfn_:bkzle4v) ([raw file](https://github.com/cockroachdb/cockroach/blob/ee2f24d5d9a7cda5b08b271de382efa9ad845155/pkg/cmd/roachtest/tests/admission_control_intent_resolution.go#L40)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

We don't need to BTW - even running prometheus on the same node, or not at all, is ok. These runs happen on GCE and we now have that global prometheus instance that knows how to scrape every endpoint. 
</blockquote></details>

I added this only because someone could do `--cloud aws`. I've left this unchanged.

**Comment by sumeerbhola** [comment]
bors r=irfansharif

**Pull request merged**